### PR TITLE
Include and parse shareWithAdditionalInfo to be used in shares autocompletion and list

### PR DESCRIPTION
**Pull request by davigonz**
Implements https://github.com/owncloud/android/issues/2284
Needed for https://github.com/owncloud/android/pull/2428

### CODE REVIEW

File: owncloudComLibrary/src/main/java/com/owncloud/android/lib/resources/shares/GetRemoteShareesOperation.java [hunk 1]
```diff
@@ -3,7 +3,7 @@
  *   @author masensio
  *   @author David A. Velasco
  *   @author David González Verdugo
- *   Copyright (C) 2018 ownCloud GmbH.
+ *   Copyright (C) 2019 ownCloud GmbH.
```
Comment (hannesa2) [inline owncloudComLibrary/src/main/java/com/owncloud/android/lib/resources/shares/GetRemoteShareesOperation.java]: cron job !

File: owncloudComLibrary/src/main/java/com/owncloud/android/lib/resources/shares/OCShare.java [hunk 2]
```diff
@@ -90,6 +91,7 @@
     private long mExpirationDate;
     private String mToken;
     private String mSharedWithDisplayName;
+    private String mSharedWithAdditionalInfo;
```
Comment (hannesa2) [inline owncloudComLibrary/src/main/java/com/owncloud/android/lib/resources/shares/OCShare.java]: Any need for an init value ? Just to think about

**Comment by hannesa2** [comment]
Is there a chance to auto generate 'OCShare.java' ?

**Comment by davigonz** [comment]
> Is there a chance to auto generate 'OCShare.java' ?

Hmm, no, why would you need that?

**Comment by hannesa2** [comment]
Just to know. I've not much knowledge about this api, is it REST style ? At least it looks like.
If it's really a json, consider to auto generate it with 
eg https://github.com/joelittlejohn/jsonschema2pojo.
Benefit: if definition is changed on backend you see it immediate if you have a gradle task during build.
-just as an idea-

File: owncloudComLibrary/src/main/java/com/owncloud/android/lib/resources/shares/ShareXMLParser.java [hunk 3]
```diff
@@ -36,413 +37,407 @@
...
+	 * @throws IOException
+	 */
+	public ArrayList<OCShare> parseXMLResponse(InputStream is) throws XmlPullParserException,
+			IOException {
+
+		try {
+			// XMLPullParser
+			XmlPullParserFactory factory = XmlPullParserFactory.newInstance();
+			factory.setNamespaceAware(true);
+
```
Comment (abelgardep) [inline owncloudComLibrary/src/main/java/com/owncloud/android/lib/resources/shares/ShareXMLParser.java]: What is the criteria of this long indentation?

File: owncloudComLibrary/src/main/java/com/owncloud/android/lib/resources/shares/GetRemoteShareesOperation.java [hunk 4]
```diff
@@ -46,64 +47,70 @@
...
+ *    url argument: itemType - string, required
+ *    url argument: format - string, optional
+ *    url argument: search - string, optional
+ *    url arguments: perPage - int, optional
+ *    url arguments: page - int, optional
+ *
  * Status codes:
- * 100 - successful
+ *    100 - successful
  *
```
Comment (abelgardep) [inline owncloudComLibrary/src/main/java/com/owncloud/android/lib/resources/shares/GetRemoteShareesOperation.java]: Why do you add those spaces?

File: owncloudComLibrary/src/main/java/com/owncloud/android/lib/resources/shares/ShareXMLParser.java [hunk 5]
```diff
@@ -36,413 +37,407 @@
...
+			XmlPullParser parser = Xml.newPullParser();
+			parser.setFeature(XmlPullParser.FEATURE_PROCESS_NAMESPACES, false);
+			parser.setInput(is, null);
+			parser.nextTag();
+			return readOCS(parser);
+
```
Comment (abelgardep) [inline owncloudComLibrary/src/main/java/com/owncloud/android/lib/resources/shares/ShareXMLParser.java]: What is the criteria of this long indentation?

File: owncloudComLibrary/src/main/java/com/owncloud/android/lib/resources/shares/GetRemoteShareesOperation.java (hunk 4 above)
Comment (davigonz) [reply owncloudComLibrary/src/main/java/com/owncloud/android/lib/resources/shares/GetRemoteShareesOperation.java]: https://github.com/owncloud/android-library/pull/224 apply a proper code style for all the files and I did not apply the proper style in here, good catch, I've just already fixed it.

File: owncloudComLibrary/src/main/java/com/owncloud/android/lib/resources/shares/ShareXMLParser.java (hunk 5 above)
Comment (davigonz) [reply owncloudComLibrary/src/main/java/com/owncloud/android/lib/resources/shares/ShareXMLParser.java]: Same here

**Comment by davigonz** [comment]
@abelgardep I've applied the changes requested, please review and approve them if you agree, thanks

**Pull request merged**